addons/account/models/account_move.py

```diff
@@ -1190,7 +1189,7 @@ def _compute_payment_state(self):
     @api.depends('payment_state', 'state')
     def _compute_status_in_payment(self):
         for move in self:
-            move.status_in_payment = move.state if move.state in ('draft', 'cancel') else move.payment_state
+            move.status_in_payment = move.state if move.state == 'cancel' or (move.state == 'draft' and move.payment_state == 'not_paid') else move.payment_state
```

Comment:
Again, I don't get your weird condition about `payment_state`. This behavior must absolutely be tested because it's barely understandable...  in `test_payment_move_state` maybe?

Reply:
Before:
posted moves will use the payment state instead of move state

After:
posted moves and (partially) paid draft moves will use the payment state instead of move state

:thinking: about the test